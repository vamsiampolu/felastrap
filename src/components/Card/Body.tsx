import { useFela } from "react-fela";
import type { DivProps, StyleProps } from "../common";
import { overlayBase, overlayLayoutRule } from "./imageStyle";

const padding = {
  paddingTop: "1rem",
  paddingBottom: "1rem",
  paddingLeft: "1rem",
  paddingRight: "1rem",
};

const body = ({ color }: StyleProps<CardBodyProps>) => ({
  ...padding,
  color,
  flexGrow: 1,
  flexShrink: 1,
  flexBasis: "auto",
});

export type CardBodyProps = DivProps & { color?: string; overlay?: boolean };

export const CardBody = ({
  className = "",
  children,
  color = "#000",
  overlay = false,
  ...other
}: CardBodyProps) => {
  const { css } = useFela({ color, overlay });
  const cx = `${css([body, overlayBase, overlayLayoutRule])} ${
    className || ""
  }`.trim();
  return (
    <div data-testid="card-body" className={cx} {...other}>
      {children}
    </div>
  );
};
