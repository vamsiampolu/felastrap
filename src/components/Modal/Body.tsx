import { useFela } from "react-fela";
import { DivProps } from "../common";

const rule = () => ({
  position: "relative",
  padding: "15px",
  borderBottom: "1px solid #dee2e6",
});

const scrollableRule = ({ scrollable = false }) =>
  scrollable
    ? {
        overflowY: "auto !important",
        minHeight: "1500px",
      }
    : {};

export const ModalBody = ({
  className = "",
  children,
  scrollable = false,
  ...props
}: DivProps & { scrollable?: boolean }) => {
  const { css } = useFela({ scrollable });
  const cx = `${css([rule, scrollableRule])} bs-modal-body ${className}`.trim();
  return (
    <div data-testid="modal-body" className={cx} {...props}>
      {children}
    </div>
  );
};
