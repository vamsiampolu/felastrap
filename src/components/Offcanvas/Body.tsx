import { useFela } from "react-fela";
import { DivProps } from "../common";

const rule = () => ({
  flexGrow: 1,
  padding: "1rem 1rem",
  overflowY: "auto",
});

export const OffcanvasBody = ({
  className = "",
  children,
  ...props
}: DivProps) => {
  const { css } = useFela();
  const cx = `${css(rule)} offcanvas-body ${className}`.trim();
  return (
    <div data-testid="offcanvas-body" className={cx} {...props}>
      {children}
    </div>
  );
};
