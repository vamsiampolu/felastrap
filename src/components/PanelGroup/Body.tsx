import { useFela } from "react-fela";
import { DivProps } from "../common";
import { Content } from "../Collapse/Content";

const flushBody = ({ flush = false }) => {
  return flush ? { borderWidth: 0 } : {};
};

const rule = () => ({
  padding: "1rem 1.25rem",
});

export type BodyProps = DivProps & { flush?: boolean };
export const Body = ({
  className = "",
  flush = false,
  children,
  ...other
}: BodyProps) => {
  const { css } = useFela({ flush });
  const cx = `${css([rule, flushBody])} panel-group-body ${className}`;
  return (
    <div data-testid="panel-group-body" className={cx} {...other}>
      {children}
    </div>
  );
};

export const BodyContainer = ({ className = "", id, ...props }: DivProps) => {
  return (
    <Content id={id}>
      <Body className={className} {...props} />
    </Content>
  );
};
