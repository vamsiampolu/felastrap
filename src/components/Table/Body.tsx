import { IStyle } from "fela";
import { ComponentPropsWithoutRef, useContext } from "react";
import { useFela } from "react-fela";
import { TableContext } from "./TableContext";
import { hoveredBackgroundColorByVariant } from "./variant";
import { Variant } from "../common";

export type TableBodyProps = ComponentPropsWithoutRef<"tbody"> & {
  striped?: boolean;
  hover?: boolean;
  bordered?: boolean;
};

const rule = () => ({ verticalAlign: "inherit" });

const hoverRule = ({ hover = false, variant = "default" }): IStyle =>
  hover
    ? {
        "> tr:hover > *": {
          cursor: "pointer",
          color: "#000",
          backgroundColor: hoveredBackgroundColorByVariant[variant as Variant],
        },
      }
    : {};

const borderedRule = ({ bordered = false }) =>
  bordered
    ? {
        borderWidth: "1px 0",
      }
    : {};

export const TableBody = ({
  className = "",
  children,
  ...props
}: TableBodyProps) => {
  const { hover, bordered, variant = "default" } = useContext(TableContext);
  const { css } = useFela({ hover, bordered, variant });
  const cx = `${css([
    rule,
    hoverRule,
    borderedRule,
  ])} table-body ${className}`.trim();
  return (
    <tbody data-testid="table-body" className={cx} {...props}>
      {children}
    </tbody>
  );
};
