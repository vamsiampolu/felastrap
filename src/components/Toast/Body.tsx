import { useFela } from "react-fela";
import { DivProps } from "../common";

const rule = () => ({
  padding: "0.75rem",
  wordWrap: "break-word",
});

export const ToastBody = ({ children, className = "", ...props }: DivProps) => {
  const { css } = useFela();
  const cx = `${css(rule)} toast-body ${className}`.trim();
  return (
    <div data-testid="toast-body" className={cx} {...props}>
      {children}
    </div>
  );
};
