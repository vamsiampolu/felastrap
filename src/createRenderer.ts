import { createRenderer as createFelaRenderer } from "fela";
import expandShorthand from "fela-plugin-expand-shorthand";
import placeholderPrefixer from "fela-plugin-placeholder-prefixer";
import prefixer from "fela-plugin-prefixer";
import sortClassNames from "fela-sort-classnames";
import sortMediaQueryMobileFirst from "fela-sort-media-query-mobile-first";

export const createRenderer = () => {
  const autoMergeLonghands = true;
  return createFelaRenderer({
    plugins: [
      prefixer(),
      placeholderPrefixer(),
      expandShorthand(autoMergeLonghands),
    ],
    enhancers: [sortClassNames(), sortMediaQueryMobileFirst()],
  });
};
