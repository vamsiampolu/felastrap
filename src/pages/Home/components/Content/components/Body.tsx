import { ComponentPropsWithoutRef, ReactNode } from "react";
import { useFela } from "react-fela";
import { IconLink } from "./IconLink";
import { ArrowRightIcon } from "../../../../../components/Icons/ArrowRightIcon";

const articleBody = () => ({
  display: "flex",
  fontSize: "1rem",
  fontWeight: 400,
  "@media((min-width: 768px)": {
    fontSize: "1.25rem",
  },
  marginBottom: 0,
  justifyContent: "flex-start",
  "@media (min-width: 768px)": {
    fontSize: "1.25rem",
  },
});

export const BodyParagraph = ({
  className = "",
  children,
  linkText,
  href,
  ...props
}: ComponentPropsWithoutRef<"p"> & { href: string; linkText: ReactNode }) => {
  const { css } = useFela();
  const cx = `${css(articleBody)} column-body ${className}`.trim();
  return (
    <p className={cx} {...props}>
      <IconLink href={href}>
        {linkText}
        <ArrowRightIcon />
      </IconLink>
      {children}
    </p>
  );
};
