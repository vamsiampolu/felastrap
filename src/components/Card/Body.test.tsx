import { IRenderer } from "fela";
import { CardBody } from "./Body";
import { createRenderer, cssReboot } from "../../test-helpers/createRenderer";
import { createSnapshot } from "../../test-helpers/createSnapshot";

describe("components/Card/Body", () => {
  let renderer: IRenderer;

  beforeEach(() => {
    renderer = createRenderer();
    cssReboot(renderer);
  });

  afterEach(() => {
    renderer.clear();
  });

  it("should render a card body", async () => {
    const component = <CardBody>Card Body</CardBody>;
    const snapshot = await createSnapshot({ component, renderer });
    expect(snapshot).toMatchSnapshot();
  });
});
