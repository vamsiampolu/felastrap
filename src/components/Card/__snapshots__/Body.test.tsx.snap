// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Card/Body should render a card body 1`] = `
".a {
  padding-top: 1rem;
}
.b {
  padding-bottom: 1rem;
}
.c {
  padding-left: 1rem;
}
.d {
  padding-right: 1rem;
}
.e {
  color: #000;
}
.f {
  flex-grow: 1;
}
.g {
  -webkit-flex-shrink: 1;
  -ms-flex-shrink: 1;
  flex-shrink: 1;
}
.h {
  -webkit-box-flex-basis: auto;
  -webkit-flex-basis: auto;
  -ms-flex-basis: auto;
  flex-basis: auto;
}


<div data-testid="card-body" class="a b c d e f g h">Card Body</div>
;
"
`;
